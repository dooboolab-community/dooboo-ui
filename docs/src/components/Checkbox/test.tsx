import {Button, Checkbox, Hr, ThemeProvider} from 'dooboo-ui';
import type {FC, ReactElement} from 'react';
import {FontAwesome, createIconSetFromIcoMoon} from '@expo/vector-icons';
import {IC_FACEBOOK, IC_GOOGLE} from '../../icon';
import {Image, View} from 'react-native';
import React, {useState} from 'react';
import styled, {css} from '@emotion/native';

import {action} from '@storybook/addon-actions';
import {useFonts} from 'expo-font';

const collectingFontIconSelection = require('./selection.json');

const Icon = createIconSetFromIcoMoon(
  collectingFontIconSelection,
  'IcoMoon',
  './doobooui.ttf',
);

const StoryContainer = styled.View`
  flex: 1;
  align-self: stretch;
`;

const StyledText = styled.Text`
  margin: 8px;
`;

const ScrollContainer = styled.ScrollView`
  width: 100%;
`;

const CheckboxDefault: FC = () => {
  const [fontsLoaded] = useFonts({
    IcoMoon: require('./doobooui.ttf'),
  });

  // if (!fontsLoaded) return <View />;

  return (
    <StoryContainer>
      <ScrollContainer
        contentContainerStyle={{
          paddingVertical: 60,
          paddingHorizontal: 40,
          justifyContent: 'center',
          alignItems: 'center',
        }}>
        <Checkbox checked={true} type="danger" />
      </ScrollContainer>
    </StoryContainer>
  );
};

const Default = (): ReactElement => (
  <ThemeProvider>
    <>
      <Icon name="like-solid" color="black" size={32} />
      <FontAwesome name="plus" />
      <CheckboxDefault />
    </>
  </ThemeProvider>
);

export default Default;
