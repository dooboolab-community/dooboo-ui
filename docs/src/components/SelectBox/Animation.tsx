import React, {useState} from 'react';
import {ThemeProvider, ThemeType, useTheme} from '@dooboo-ui/theme';

import type {FC} from 'react';
import {SelectBox} from 'dooboo-ui';
import {View} from 'react-native';
import styled from '@emotion/native';
import {useFonts} from 'expo-font';

const Container = styled.View`
  flex: 1;
  align-self: stretch;
  background-color: ${({theme}) => theme.background};

  align-items: center;
  justify-content: center;
`;

const StyledText = styled.Text`
  color: ${({theme}) => theme.text};
  font-size: 16px;
  margin: 20px;
`;

function SelectBoxStory(): React.ReactElement {
  const data = ['Item1', 'Item2', 'Item3', 'Item4'];

  const {
    media: {isMobile},
  } = useTheme();

  const [selectedIndex, setSelectedIndex] = useState(0);
  const [count, setCount] = useState(0);

  const [fontsLoaded] = useFonts({
    IcoMoon: require('../../assets/doobooui.ttf'),
  });

  const handlePressIncrease = (): void => {
    setCount(count + 1);
  };

  if (!fontsLoaded) return <View />;

  return (
    <Container style={{flexDirection: isMobile ? 'column' : 'row'}}>
      <View
        style={{
          flexDirection: 'row',
          flexWrap: 'wrap',
          justifyContent: 'center',
          margin: 10,
        }}
      >
        <SelectBox
          data={data}
          onSelect={(_, index) => setSelectedIndex(index)}
          selectedIndex={selectedIndex}
          rotateAnimDuration={1000}
          onPress={handlePressIncrease}
          style={{padding: 15}}
        />
        <StyledText>press count : {count}</StyledText>
      </View>
    </Container>
  );
}

export const Animation: FC<{themeType: ThemeType}> = ({themeType}) => {
  return (
    <ThemeProvider initialThemeType={themeType}>
      <SelectBoxStory />
    </ThemeProvider>
  );
};
