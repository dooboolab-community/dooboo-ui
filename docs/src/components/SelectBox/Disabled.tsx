import React, {useState} from 'react';
import {SelectBox, ThemeProvider, ThemeType, useTheme} from 'dooboo-ui';

import type {FC} from 'react';
import {View} from 'react-native';
import styled from '@emotion/native';
import {useFonts} from 'expo-font';

const Container = styled.View`
  flex: 1;
  align-self: stretch;
  background-color: ${({theme}) => theme.background};

  align-items: center;
  justify-content: center;
`;

function SelectBoxStory(): React.ReactElement {
  const data = ['Item1', 'Item2', 'Item3', 'Item4'];

  const {
    media: {isMobile},
  } = useTheme();

  const [selectedIndex, setSelectedIndex] = useState(0);

  const [fontsLoaded] = useFonts({
    IcoMoon: require('../../assets/doobooui.ttf'),
  });

  if (!fontsLoaded) return <View />;

  return (
    <Container style={{flexDirection: isMobile ? 'column' : 'row'}}>
      <SelectBox
        data={data}
        onSelect={(_, index) => setSelectedIndex(index)}
        selectedIndex={selectedIndex}
        disabled={true}
        style={{margin: 10}}
      />
    </Container>
  );
}

export const Disabled: FC<{themeType: ThemeType}> = ({themeType}) => {
  return (
    <ThemeProvider initialThemeType={themeType}>
      <SelectBoxStory />
    </ThemeProvider>
  );
};
