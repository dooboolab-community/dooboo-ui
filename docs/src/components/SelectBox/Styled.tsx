import React, {useState} from 'react';
import {ThemeProvider, ThemeType, useTheme} from '@dooboo-ui/theme';

import type {FC} from 'react';
import {SelectBox} from 'dooboo-ui';
import {View} from 'react-native';
import styled from '@emotion/native';
import {useFonts} from 'expo-font';

const Container = styled.View`
  flex: 1;
  align-self: stretch;
  background-color: ${({theme}) => theme.background};

  align-items: center;
  justify-content: center;
`;

function SelectBoxStory(): React.ReactElement {
  const data = ['Item1', 'Item2', 'Item3', 'Item4'];

  const {
    theme,
    media: {isMobile},
  } = useTheme();

  const [selectedIndex, setSelectedIndex] = useState(0);

  const [fontsLoaded] = useFonts({
    IcoMoon: require('../../assets/doobooui.ttf'),
  });

  if (!fontsLoaded) return <View />;

  return (
    <Container style={{flexDirection: isMobile ? 'column' : 'row'}}>
      <SelectBox
        data={data}
        onSelect={(_, index) => setSelectedIndex(index)}
        selectedIndex={selectedIndex}
        style={{margin: 10}}
        styles={{
          titleContainer: {
            backgroundColor: theme.success,
            borderTopWidth: 0,
            borderBottomWidth: 0,
            borderLeftWidth: 0,
            borderRightWidth: 0,
            borderRadius: 40,
            height: 40,
          },
          itemContainer: {
            borderTopWidth: 0,
            borderBottomWidth: 0,
            borderLeftWidth: 0,
            borderRightWidth: 0,
            backgroundColor: theme.paper,
            height: 40,
          },
          rightElementContainer: {
            right: 15,
          },
        }}
      />
    </Container>
  );
}

export const Styled: FC<{themeType: ThemeType}> = ({themeType}) => {
  return (
    <ThemeProvider initialThemeType={themeType}>
      <SelectBoxStory />
    </ThemeProvider>
  );
};
