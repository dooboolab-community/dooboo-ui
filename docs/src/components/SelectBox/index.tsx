import React, {useState} from 'react';
import {SelectBox, ThemeProvider, ThemeType, useTheme} from 'dooboo-ui';

import type {FC} from 'react';
import {View} from 'react-native';
import styled from '@emotion/native';
import {useFonts} from 'expo-font';

const Container = styled.View`
  flex: 1;
  align-self: stretch;
  background-color: ${({theme}) => theme.background};

  align-items: center;
  justify-content: center;
`;

const StyledText = styled.Text`
  color: ${({theme}) => theme.text};
  font-size: 16px;
  margin: 20px;
`;

function SelectBoxStory(): React.ReactElement {
  const data = ['Item1', 'Item2', 'Item3', 'Item4'];

  const {
    media: {isMobile},
  } = useTheme();

  const [selectedIndex, setSelectedIndex] = useState(0);

  const [fontsLoaded] = useFonts({
    IcoMoon: require('../../assets/doobooui.ttf'),
  });

  if (!fontsLoaded) return <View />;

  return (
    <Container style={{flexDirection: isMobile ? 'column' : 'row'}}>
      <SelectBox
        data={data}
        onSelect={(_, index) => setSelectedIndex(index)}
        selectedIndex={selectedIndex}
        style={{margin: 10}}
      />
      <StyledText>selectedIndex : {selectedIndex}</StyledText>
    </Container>
  );
}

export const Basic: FC<{themeType: ThemeType}> = ({themeType}) => {
  return (
    <ThemeProvider initialThemeType={themeType}>
      <SelectBoxStory />
    </ThemeProvider>
  );
};
