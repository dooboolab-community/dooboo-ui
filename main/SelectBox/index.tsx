import {
  Animated,
  Easing,
  Platform,
  StyleProp,
  TextStyle,
  TouchableOpacity,
  View,
  ViewStyle,
} from 'react-native';
import React, {ReactElement, useEffect, useRef, useState} from 'react';

import {DoobooTheme} from '../theme';
import {Icon} from '../Icon';
import {Typography} from '../Typography';
import styled from '@emotion/native';
import {withTheme} from '@emotion/react';

const Container = styled.View`
  z-index: 999;
`;

const SelectedItem = styled.View`
  width: 200px;
  height: 30px;
  border-width: 1px;
  border-color: ${({theme}) => theme.primary};
  background-color: ${({theme}) => theme.textContrast};

  flex-direction: row;
  justify-content: center;
  align-items: center;
`;

const Item = styled.View`
  width: 200px;
  height: 30px;
  border-bottom-width: 1px;
  border-left-width: 1px;
  border-right-width: 1px;
  border-color: ${({theme}) => theme.primary};
  background-color: ${({theme}) => theme.textContrast};

  justify-content: center;
  align-items: center;
`;

interface ItemCompProps {
  value: string;
  translateYValue: number;
  onPress: (value: string) => void;
}

function ItemComp({
  value,
  translateYValue,
  onPress: setValue,
}: ItemCompProps): ReactElement {
  return (
    <TouchableOpacity onPress={() => setValue(value)} activeOpacity={1}>
      <Item
        style={{
          position: 'absolute',
          transform: [{translateY: translateYValue}],
        }}>
        <Typography.Body2>{value}</Typography.Body2>
      </Item>
    </TouchableOpacity>
  );
}

type Styles = {
  container?: StyleProp<ViewStyle>;
  itemContainer?: StyleProp<ViewStyle>;
  selectedText?: StyleProp<TextStyle>;
  itemText?: StyleProp<TextStyle>;
};

interface Props {
  data: string[];
  theme?: DoobooTheme;
  style?: StyleProp<ViewStyle>;
  styles?: Styles;
  onPress?: () => void;
  rightElement?: ReactElement;
}

function Component({
  style,
  styles,
  data,
  rightElement = <Icon name="chevron-down-light" />,
}: Props): ReactElement {
  const ITEM_HEIGHT = 30;
  const [isOpened, setIsOpened] = useState(false);
  const [selectedValue, setSelectedValue] = useState<string | null>(null);

  const rotateAnimValue = useRef(new Animated.Value(0)).current;

  const handlePress = (): void => {
    setIsOpened((prev) => !prev);
  };

  const handleSelectedValue = (value: string): void => {
    setSelectedValue(value);
  };

  useEffect(() => {
    const toValue = isOpened ? 1 : 0;

    Animated.timing(rotateAnimValue, {
      toValue,
      duration: 200,
      easing: Easing.linear,
      useNativeDriver: Platform.OS !== 'web' ? true : false,
    }).start();
  }, [isOpened, rotateAnimValue]);

  return (
    <>
      <Container>
        <TouchableOpacity onPress={handlePress} activeOpacity={1}>
          <SelectedItem>
            <Typography.Body2>{selectedValue ?? data[0]}</Typography.Body2>
            <Animated.View
              style={{
                position: 'absolute',
                right: 10,
                transform: [
                  {
                    rotate: rotateAnimValue.interpolate({
                      inputRange: [0, 1],
                      outputRange: ['0deg', '180deg'],
                    }),
                  },
                ],
              }}>
              {rightElement}
            </Animated.View>
          </SelectedItem>
        </TouchableOpacity>
        <View>
          {isOpened &&
            data.map((datum, key) => (
              <ItemComp
                value={datum}
                key={key}
                onPress={handleSelectedValue}
                translateYValue={ITEM_HEIGHT * key}
              />
            ))}
        </View>
      </Container>
    </>
  );
}

export const SelectBox = withTheme(Component);
