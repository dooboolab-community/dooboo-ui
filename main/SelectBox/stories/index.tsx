import React, {ReactElement} from 'react';

import SelectBoxStory from './DefaultStory';
import {ThemeProvider} from '@dooboo-ui/theme';
import {View} from 'react-native';
import {storiesOf} from '@storybook/react-native';

/**
 * Below are stories for web
 */
export default {
  title: 'LoadingIndicator',
};

export const toStorybook = (): ReactElement => <SelectBoxStory />;

toStorybook.story = {
  name: 'default',
};

/**
 * Below are stories for app
 */
storiesOf('SelectBox', module)
  .add('SelectBox - light', () => (
    <ThemeProvider initialThemeType="light">
      <SelectBoxStory />
    </ThemeProvider>
  ))
  .add('SelectBox - dark', () => (
    <View
      style={{
        flex: 1,
        backgroundColor: 'black',
        alignSelf: 'stretch',
        alignItems: 'center',
        justifyContent: 'center',
      }}
    >
      <ThemeProvider initialThemeType="dark">
        <SelectBoxStory />
      </ThemeProvider>
    </View>
  ));
