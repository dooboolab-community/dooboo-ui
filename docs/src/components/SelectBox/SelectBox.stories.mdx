import {Meta, Story, Canvas} from '@storybook/addon-docs/blocks';
import {ThemeProvider} from '@dooboo-ui/theme';

import {Basic} from './index.tsx';
import {Disabled} from './Disabled.tsx';
import {Animation} from './Animation.tsx';
import {Styled} from './Styled.tsx'

<Meta title="Components/SelectBox" />

# SelectBox

> [SelectBox] component can be used to select specific value from array data. It can take a string array or object array.

## Usage

```tsx
import {SelectBox} from 'dooboo-ui';

<SelectBox {...props} />;
```

## Props

| props               | necessary | types                                                  |               default               | description                                             |
| ------------------- | :-------: | ------------------------------------------------------ | :---------------------------------: | ------------------------------------------------------- |
| theme               |           | `DoobooTheme`                                          |                                     |                                                         |
| data                |     ✓     | T[]                                                    |                                     | The data type can be used to intenal generic type       |
| selectedIndex       |     ✓     | number                                                 |                 `0`                 | Initial selectedIndex                                   |
| onSelect            |     ✓     | (item: T, index: number) => void                       |                                     | Triggered when you select item                          |
| onPress             |           | () => void                                             |                                     | Triggered when you press title                          |
| style               |           | `StyleProps<ViewStyle>`                                |                                     |                                                         |
| styles              |           | `Styles`                                               |             `undefined`             |                                                         |
| disabled            |           | boolean                                                |               `false`               | Touch event cannot be triggered when this props is true |
| rightElement        |           | ReactElement \| null                                   | `<Icon name="chevron-down-light"/>` |                                                         |
| rotateAnimDuration  |           | number                                                 |                `200`                | Duration time for Rotating Animation                    |
| isRightElemAnimated |           | boolean                                                |               `true`                | Rotating Animation for right element                    |
| titleTouchableProps |           | Partial<TouchableOpacityProps & GenericTouchableProps> |        `{activeOpacity: 1} `        |                                                         |
| itemTouchableProps  |           | Partial<TouchableOpacityProps & GenericTouchableProps> |        `{activeOpacity: 1} `        |                                                         |

### Basic Style
`data` `onSelect` `selectedIndex` props are necessary.
User can set items of select box by setting `data` array and represent index in `data` array of selected item by using `selectedIndex` prop.

```tsx
function SelectBoxStory(): React.ReactElement {
  const data = ['Item1', 'Item2', 'Item3', 'Item4'];
  const [selectedIndex, setSelectedIndex] = useState(0);

  return (
    <Container>
      <SelectBox
        data={data}
        onSelect={(_, index) => setSelectedIndex(index)}
        selectedIndex={selectedIndex}
      />
    </Container>
  );
}
```
<Basic themeType="light"/>
<Basic themeType="dark"/>

### Disabled SelectBox
The `disabled` prop can make disabled selectbox that user can't select items.

```tsx
function SelectBoxStory(): React.ReactElement {
  const data = ['Item1', 'Item2', 'Item3', 'Item4'];

  const [selectedIndex, setSelectedIndex] = useState(0);

  return (
    <Container>
      <SelectBox
        data={data}
        onSelect={(_, index) => setSelectedIndex(index)}
        selectedIndex={selectedIndex}
        disabled={true}
      />
    </Container>
  );
}
```
<Disabled themeType="light"/>
<Disabled themeType="dark"/>


### Animation and Event
`rotateAnimDuration` `onPress`

The `rotateAnimDuration` prop can adjust time the animation of right element operates.

The `onPress` prop triggers event when users click title of select box.

```tsx
function SelectBoxStory(): React.ReactElement {
  const data = ['Item1', 'Item2', 'Item3', 'Item4'];
  const [selectedIndex, setSelectedIndex] = useState(0);
  const [count, setCount] = useState(0);

  const handlePressIncrease = () => {
    setCount(count + 1);
  }

  return (
    <Container>
      <View
        style = {{
          flexDirection: 'row',
          flexWrap: 'wrap',
          justifyContent: 'center',
        }}>
        <SelectBox
            data={data}
            onSelect={(_, index) => setSelectedIndex(index)}
            selectedIndex={selectedIndex}
            rotateAnimDuration={1000}
            onPress={handlePressIncrease}
            style={{padding: 15}}
        />
        <StyledText>press count : {count}</StyledText>
      </View>
    </Container>
  );
}
```
<Animation/>

### Styled SelectBox
The `styles` `style` props can make `SelectBox` custom as users want.

```tsx
function SelectBoxStory(): React.ReactElement {
  const data = ['Item1', 'Item2', 'Item3', 'Item4'];
  const [selectedIndex, setSelectedIndex] = useState(0);

  return (
    <Container>
      <SelectBox
        data={data}
        onSelect={(_, index) => setSelectedIndex(index)}
        selectedIndex={selectedIndex}
        styles={{
          titleContainer: {
            backgroundColor: theme.success,
            borderTopWidth: 0,
            borderBottomWidth: 0,
            borderLeftWidth: 0,
            borderRightWidth: 0,
            borderRadius: 40,
            height: 40,
          },
          itemContainer: {
            borderTopWidth: 0,
            borderBottomWidth: 0,
            borderLeftWidth: 0,
            borderRightWidth: 0,
            backgroundColor: theme.paper,
            height: 40,
          },
          rightElementContainer: {
            right: 15,
          },
        }}
      />
    </Container>
  );
}
```
<Styled themeType="light"/>
<Styled themeType="dark"/>