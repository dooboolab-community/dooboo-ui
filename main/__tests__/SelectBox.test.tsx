import React, {ReactElement} from 'react';
import {RenderAPI, fireEvent, render} from '@testing-library/react-native';
import {createComponent, createTestProps} from '../../test/testUtils';

import {SelectBox} from '../SelectBox';

let props: any;
let component: ReactElement;
let testingLib: RenderAPI;

const data = ['item1', 'item2', 'item3', 'item4'];

describe('[SelectBox] render test', () => {
  it('should render without crashing', () => {
    props = createTestProps({data});
    component = createComponent(<SelectBox {...props} />);
    testingLib = render(component);

    const json = testingLib.toJSON();

    expect(json).toMatchSnapshot();
  });

  it('should rightElement rotate when isRotate is true', () => {
    props = createTestProps({
      isRotate: true,
      data,
    });

    component = createComponent(<SelectBox {...props} />);

    testingLib = render(component);

    const json = testingLib.toJSON();

    expect(json).toMatchSnapshot();
  });

  it('should render first value of data at first', () => {
    props = createTestProps({data});

    component = createComponent(<SelectBox {...props} />);

    testingLib = render(component);

    const selectedValue = testingLib.getByTestId('selected-value');

    expect(selectedValue.props.children).toEqual('item1');
  });

  it('should adjust duration of animation depends on rotateDuration props value', () => {
    props = createTestProps({
      rotateDuration: 500,
      data,
    });

    component = createComponent(<SelectBox {...props} />);
    testingLib = render(component);

    const json = testingLib.toJSON();

    expect(json).toMatchSnapshot();
  });
});
