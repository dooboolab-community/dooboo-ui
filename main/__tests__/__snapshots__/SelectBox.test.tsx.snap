// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[SelectBox] render test should adjust duration of animation depends on rotateDuration props value 1`] = `
<View
  style={
    Object {
      "zIndex": 999,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderLeftWidth": 1,
            "borderRightWidth": 1,
            "borderTopWidth": 1,
            "flexDirection": "row",
            "height": 30,
            "justifyContent": "center",
            "width": 200,
          },
          Array [
            Object {
              "backgroundColor": "white",
              "borderColor": "black",
            },
            undefined,
          ],
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 14,
            },
            Object {
              "includeFontPadding": false,
            },
          ]
        }
        testID="selected-value"
      >
        item1
      </Text>
      <View
        style={
          Object {
            "position": "absolute",
            "right": 10,
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
            ],
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "black",
                },
                undefined,
              ],
              Object {
                "fontFamily": "IcoMoon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <View />
</View>
`;

exports[`[SelectBox] render test should render without crashing 1`] = `
<View
  style={
    Object {
      "zIndex": 999,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderLeftWidth": 1,
            "borderRightWidth": 1,
            "borderTopWidth": 1,
            "flexDirection": "row",
            "height": 30,
            "justifyContent": "center",
            "width": 200,
          },
          Array [
            Object {
              "backgroundColor": "white",
              "borderColor": "black",
            },
            undefined,
          ],
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 14,
            },
            Object {
              "includeFontPadding": false,
            },
          ]
        }
        testID="selected-value"
      >
        item1
      </Text>
      <View
        style={
          Object {
            "position": "absolute",
            "right": 10,
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
            ],
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
  <View />
</View>
`;

exports[`[SelectBox] render test should rightElement rotate when isRotate is true 1`] = `
<View
  style={
    Object {
      "zIndex": 999,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderLeftWidth": 1,
            "borderRightWidth": 1,
            "borderTopWidth": 1,
            "flexDirection": "row",
            "height": 30,
            "justifyContent": "center",
            "width": 200,
          },
          Array [
            Object {
              "backgroundColor": "white",
              "borderColor": "black",
            },
            undefined,
          ],
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 14,
            },
            Object {
              "includeFontPadding": false,
            },
          ]
        }
        testID="selected-value"
      >
        item1
      </Text>
      <View
        style={
          Object {
            "position": "absolute",
            "right": 10,
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
            ],
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "black",
                },
                undefined,
              ],
              Object {
                "fontFamily": "IcoMoon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <View />
</View>
`;
